File: labs/lab9/q4.py
#Code by Mudit
#Date: 23/09/2026
import numpy as np

def firstStable(a, tolerance):
    differences = np.abs(np.diff(a))  # Finds absolute successive differences
    stable = np.where(differences <= tolerance)[0]  # Conditions

    if len(stable) > 0:
        return stable[0]  # Returns the first stable index

    return -1  # No stable index found


a = np.array(list(map(float, input().split())))  # Takes readings
tolerance = float(input())  # Takes tolerance

print(firstStable(a, tolerance))

File: labs/lab9/q4.c
//Code by Mudit
//Date: 23/09/2026
#include <stdio.h>
#include <math.h>

//Definig the function given in the question
int firstStable(double a[], int n, double tolerance)
{   
    //Checking one by one
    for (int i = 0; i < n - 1; i++)
    {
        if (fabs(a[i + 1] - a[i]) <= tolerance)
            return i;
    }
    
    //If no such found return -1
    return -1;
}

int main()
{
    int n;
    double tolerance;
    
    //Taking input
    printf("Enter how many numbers you want to enter: ");
    scanf("%d", &n);

    double a[n];
    printf("Enter the values: ");

    for (int i = 0; i < n; i++)
        scanf("%lf", &a[i]);

    printf("Enter tolerance: ");
    scanf("%lf", &tolerance);

    //Using the function
    printf("Output is %d\n", firstStable(a, n, tolerance));

    return 0;
}
